drivers/dai/intel/ssp/ssp.c

```diff
@@ -0,0 +1,2017 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <spinlock.h>
+#include <devicetree.h>
+#define LOG_DOMAIN sof_dai_ssp
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include "ssp.h"
+#include "dai-params-intel.h"
+#include "dai-params-intel-ipc4.h"
+
+#define DT_DRV_COMPAT intel_ssp_dai
+
+#define dai_set_drvdata(dai, data) (dai->priv_data = data)
+#define dai_get_drvdata(dai) dai->priv_data
+#define dai_get_mn(dai) dai->plat_data.mn_inst
+#define dai_get_ftable(dai) dai->plat_data.ftable
+#define dai_get_fsources(dai) dai->plat_data.fsources
+#define dai_mn_base(dai) dai->plat_data.mn_inst->base
+#define dai_base(dai) dai->plat_data.base
+
+#define DAI_DIR_PLAYBACK	0
+#define DAI_DIR_CAPTURE		1
+#define SSP_ARRAY_INDEX(dir) dir == DAI_DIR_RX ? DAI_DIR_CAPTURE : DAI_DIR_PLAYBACK
+
+/* stream PCM frame format */
+enum sof_ipc_frame {
```

Comment:
This should be define in either the IPC3/4 header as it's configured externally by clients ?

Reply:
removed